ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -772,6 +775,183 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
     }
   }
 
+  @Override
+  public StoreBatchDeleteInfo batchDelete(List<MessageInfo> infosToDelete) throws StoreException {
+    checkStarted();
+    checkDuplicates(infosToDelete);
+    StoreBatchDeleteInfo storeBatchDeleteInfo = new StoreBatchDeleteInfo(
+        replicaId.getPartitionId(), new ArrayList<>());
+    final Timer.Context context = metrics.batchDeleteResponse.time();
+    int i = 0;
+    List<MessageInfo> filteredInfos = null;
+    try {
+      List<IndexValue> indexValuesPriorToDelete = new ArrayList<>();
+      List<IndexValue> originalPuts = new ArrayList<>();
+      List<Short> lifeVersions = new ArrayList<>();
+      Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      filteredInfos = new ArrayList<>();
+      for (MessageInfo info : infosToDelete) {
```

Comment:
This corresponds to `infoList` which further corresponds to `convertedStoreKeys` in calling method in `AmbryRequests` class.
Do we have a upper bound on how big this list can be? 

Per design, there needs to be an upper bound to these batches.